overlord/snapstate/snapmgr.go

```diff
@@ -229,6 +232,18 @@ func (m *SnapManager) doPrepareSnap(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	// check if we need to read .sideinfo files
+	var si snap.SideInfo
+	if (ss.Flags & WithSideInfo) > 0 {
+		metafn := ss.SnapPath + ".sideinfo"
+		if j, err := ioutil.ReadFile(metafn); err == nil {
```

Comment:
to be clear what I'm trying to solve with that is the needing to open the snap blob so early where we would have to verify it. So the goal is not to open it in firstboot.go and not have to pass the name into InstallPathWithSideInfo